src/main/java/ch/njol/skript/expressions/ExprEntityStorageEntiyCount.java

```diff
@@ -0,0 +1,71 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.bukkitutil.EntityBlockStorageUtils.EntityBlockStorageType;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.block.Block;
+import org.bukkit.block.EntityBlockStorage;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Entity Storage Entity Count")
+@Description({
+	"Get the number of entities stored inside an entity block storage (i.e. beehive).",
+	"Using a specific block storage type will restrict the blocks provided to match the type."
+})
+@Examples("broadcast the beehive storage entity count of {_beehive}")
+@Since("INSERT VERSION")
+public class ExprEntityStorageEntiyCount extends PropertyExpression<Block, Integer> {
+
+	// Future proofing for any EntityBlockStorage added later on
+
+	private static final EntityBlockStorageType[] ENTITY_BLOCK_STORAGE_TYPES = EntityBlockStorageType.values();
+
+	static {
+		String[] patterns = new String[ENTITY_BLOCK_STORAGE_TYPES.length];
+		for (EntityBlockStorageType type : ENTITY_BLOCK_STORAGE_TYPES) {
+			patterns[type.ordinal()] = "[the] " + type.getCodeName() + " entity (count|amount) [of %blocks%]";
```

Comment:
cant this also just not include the specific storage type? the return method is still getEntityCount, whether its a beehive or another implementation

Reply:
I mean sure, but the idea behind it, is it being user specific/defined. I know as of currently the only extended class is Beehive, so there really is no difference in one or the other. But think of it like when grabbing entities, you can do `all entities` or `all wolves` aa  better example, `all wolves` or `all black wolves` 
By using the `entity block storage` pattern, no matter what block, as long as it extends EntityBlockStorage, it will be used. But using a specific storage, the block has to be part of that specific storage.
That was my idea anyways

Reply:
Cant it just be optional then?

Reply:
What do you mean? Can what be optional? As of currently is doesnt matter which pattern is used because theres only one EntityBlockStorage. But when something is added, then it will matter.

Reply:
Yes but cant you just make it so you do `entity count of %blocks%`. Why specify its the beehive specifically? Other blocks dont have an entity count right?

Reply:
Ohhh that's what you meant. I guess I can do that, but then the syntax would seem weird, imo.
`broadcast entity count of {_block}`
I would like to see what others say.